src/lib/builtins/array.rs

```diff
@@ -998,4 +1031,31 @@ mod tests {
         assert_eq!(forward(&mut engine, "many2.length"), "1");
         assert_eq!(forward(&mut engine, "many3.length"), "0");
     }
+
+    #[test]
+    fn reduce() {
+        let realm = Realm::create();
+        let mut engine = Executor::new(realm);
+        let init = r#"
+        function callback(acc, val) {
+            return 5;
+        }
+
+        var empty = [];
```

Comment:
Add a test for calling `reduce` on an empty `Array` with no initial value.

Reply:
Done!

Reply:
According to the spec:
> It is a TypeError if the array contains no elements and initialValue is not provided.

---

So, you can comment out the test and add a _TODO_ but the test _should not_ expect `undefined`.